pkg/services/anonymous/anonimpl/impl.go

```diff
@@ -143,6 +144,7 @@ func (a *AnonDeviceService) TagDevice(ctx context.Context, httpReq *http.Request
 	err = a.tagDeviceUI(ctx, httpReq, taggedDevice)
 	if err != nil {
 		a.log.Debug("Failed to tag device for UI", "error", err)
+		return err
```

Comment:
Question: should we check if the error is `ErrDeviceLimitReached` same as you do [here](https://github.com/grafana/grafana/pull/79265/files#diff-8b7b738288661f230305c7b8bb118ac2fdf04c5a4c804b89bdaf22b7b2e4ce7eR45) or do you want to return all errors in this case?